core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -910,6 +915,38 @@ public void setDefaultTTL(int defaultTTL) {
     }
 
     @Override
+    public Observable<DataPoint<RatioMap>> findAvailabilityRatioMap(
+            String tenantId, List<String> metrics, long start, long end, int limit, Order order) {
+        checkArgument(isValidTimeRange(start, end), "Invalid time range");
+        Observable<DataPoint<RatioMap>> result = Observable.from(metrics)
+                .map(id -> new MetricId<>(tenantId, MetricType.AVAILABILITY, id))
+                .flatMap(this::findMetric)
+                .map(metric -> findDataPoints(metric.getMetricId(), start, end, limit, order))
+                .compose(o -> buildRatioMapSeries(o, order, AvailabilityType::getText));
+        if (limit > 0) {
+            // Limit already applied on C* query, but might be exceeded after aggregation
+            // So it's applied again
+            result = result.limit(limit);
+        }
+        return result;
+    }
+
+    @Override
+    public Observable<DataPoint<RatioMap>> findAvailabilityRatioMap(
```

Comment:
You mean like the regex match?

I've built it on the same model than the existing "raw" endpoint on avail, so for now there's no such thing. But it can be interesting, to search for several metrics based on a pattern. I'll have a look at it.